### ISSUE TITLE
[core] Redaction before/after proposals + blind-eval mining (Phase 4 of #119)

### ISSUE THREAD

**Issue by mskrado** [mskrado/diagnostic-agent#126]
Phase 4 of #119. Builds on the deterministic draft (#122) and LLM draft (#124).

Phase 2 already proposes `redaction.yaml` rules from the scan census (match counts). This phase completes the redaction story and adds blind-eval case mining from real incidents.

### 1. Redaction proposals with before/after evidence

Enhance the redaction draft so every candidate rule carries a **before → after** sample taken from real (pre-scrub) lines the rule would match. The operator reviews evidence, not a suggestion.

- Keep additive-only, never remove a rule.
- High false-positive patterns (UUID, long digit runs) stay commented out with their diffs visible.
- Emit a short `redaction-review.md` (or a section of the draft report) listing each candidate, match count, and 1–3 before/after examples.
- Still never write unscrubbed secrets into the review: show the match with the sensitive span replaced by a visible marker in the "before" side too when needed, or show only enough context that the pattern is obvious without re-exposing a full token. Prefer: before line with the match bracketed, after line fully scrubbed.

### 2. Blind-eval case mining from audit logs

`diag mine-eval` reads redacted audit JSONL (`AGENT_AUDIT_LOG_DIR` / `--audits PATH`) and drafts candidate `blind_eval.yaml` cases:

| Case field | Source |
|---|---|
| `alert` labels | `report.alert` / detect state |
| `logs` | redacted log lines from the report (already scrubbed at write time) |
| `metrics` | prometheus snapshot when present |
| `expected.root_cause` | primary hypothesis / diagnosis text (operator-editable draft) |
| `expected.must_reference` | distinctive tokens that appear in the mined logs (so lint grounding can pass) |
| `expected.cause_keywords` | tokens from alertname / service / diagnosis |
| `system` | best-effort from service kind or alertname |

Safety:
- Only mine already-redacted audit lines; re-run the built-in scrubber before writing.
- Cases are written under `blind_eval.draft.yaml` (or staging) with a header that they are candidates — do not overwrite a host's curated `blind_eval.yaml` without `--force`.
- Skip records with fewer than N log lines or with no diagnosis.
- Deduplicate by alertname + log fingerprint.

### CLI

```bash
diag draft …                 # redaction section now includes before/after
diag mine-eval --audits ./audit -o ./diag-draft/blind_eval.draft.yaml
```

`--llm` is not required; mining is deterministic from the audit record.

## Acceptance criteria

- [ ] Redaction draft/report shows before→after samples per candidate rule.
- [ ] `diag mine-eval` against a fixture audit JSONL produces a valid draft dataset whose `must_reference` tokens appear in its logs (`diag lint` grounding would pass if adopted).
- [ ] Mining never writes a line that still matches a built-in secret pattern without scrubbing.
- [ ] Does not clobber an existing `blind_eval.yaml` without `--force`.
- [ ] Unit tests cover diffs, mining, scrubbing, refuse-to-clobber.
- [ ] `docs/DRAFT.md` (and a short `docs/MINE_EVAL.md` or section) documents both.

## Depends on

#122 / #123 (draft), #124 / #125 (LLM draft). Stacked on `feature/draft-llm-124`.
